docs/source/kedro_project_setup/code_beyond_starter_files.md

```diff
@@ -0,0 +1,136 @@
+# Adding code beyond starter files
+
+After you [create a Kedro project](../get_started/new_project.md) and
+[add a pipeline](../tutorial/create_a_pipeline.md), you notice that Kedro generates a
+few boilerplate files: `nodes.py`, `pipeline.py`, `pipeline_registry.py`...
+
+While those may be sufficient for a small project, they quickly become large, hard to
+read and collaborate on as your codebase grows.
+Those files also sometimes make new users think that Kedro requires code
+to be located in those exact starter files, which is not true.
+
+This section elaborates on what the Kedro requirements are for organising your project code
+in files and modules.
+It also provides examples of common scenarios such as sharing utilities between
+pipelines and using Kedro in a monorepo setup.
+
+## Where does Kedro look for code to be located
+
+The only technical constraint for arranging code in the project is that `pipeline_registry.py`
+and `settings.py` files must be located in the `<your_project>/src/<your_project>` directory, which is where
+they are created by default.
+
+`pipeline_registry.py` must have a `register_pipelines()` function that returns a `dict[str, Pipeline]`
+mapping from pipeline name to a corresponding `Pipeline` object.
+
+Other than that, **Kedro does not impose any constraints on where you should keep files with
+`Pipeline`s, `Node`s, or functions wrapped by `node`**.
+
+This being the only constraint means that you can, for example:
+* Add `utils.py` file to a pipeline folder and import utilities used by multiple
+  functions in `nodes.py` from there.
+* [Share modules between pipelines](#sharing-modules-between-pipelines).
+  Those could be utility functionalities, or a standalone module responsible for
+  the domain-specific logic.
+* [Use Kedro in a monorepo setup](#kedro-project-in-a-monorepo-setup) if there are
+  software components independent of Kedro that you want to keep together in the version control system.
+* Delete or rename a default `nodes.py` file, split it into multiple files or modules.
+* If you have multiple large `Pipeline` objects defined in a single `pipeline.py`,
+  split them into separate `.py` files. For example, in `data_processing` pipeline
+  you may want to have `cleaning_pipeline.py` and `merging_pipeline.py`.
+* Instead of registering many pipelines in `register_pipelines()` function one by one,
+  create a few `dict[str, Pipeline]` objects in different places of the project
+  and then make `register_pipelines()` return a union of those.
+
+  ```{note}
+  While Kedro features [`find_pipelines()` functionality for autodiscovery of pipelines](../nodes_and_pipelines/pipeline_registry.md#pipeline-autodiscovery),
+  for large projects you may want a finer control and register pipelines manually.
```

Comment:
> why this is the case

Because sometimes you may want to use a more explicit name than `create_pipeline`, which is what `find_pipelines()` is looking for. "Sometimes" here is a situation when a single pipeline module needs to expose different pipelines. E.g. your pipeline module is called `profiling` and it exposes `profile_table` and `profile_ml_model` pipelines, something like that.

> is it sufficient if find_pipeline search for broader pattern

I guess it's a conversation for a separate issue on `find_pipelines()` potential extension?